src/eq_schema/schema/Answer/index.js

```diff
@@ -131,41 +161,27 @@ class Answer {
       if (isNil(previousAnswer)) {
         return;
       }
-      return { answer_id: `answer${previousAnswer.id}` };
+      return {
+        value: {
+          source: "answers",
+          identifier: `answer${previousAnswer}`
+        }
+      };
     }
 
     if (entityType === "Metadata") {
       if (isNil(metadata)) {
         return;
       }
-      return { meta: metadata.key };
-    }
-    return;
-  }
 
-  static buildDateValidation(validationRule) {
-    const { enabled } = validationRule;
-    if (!enabled) {
-      return;
-    }
-
-    const comparator = Answer.buildComparator(validationRule);
-
-    if (isNil(comparator)) {
-      return;
+      return {
+        value: {
+          source: "metadata",
+          identifier: getMetadata(ctx, metadata).key
```

Comment:
This was the issue. Author isn't returning the metadata key. It's returning the ID of the metadata which prevents runner from doing a metadata lookup to find the correct value.

The alternative solution would be to change the ouput from Author so the context doesn't need to be drilled down into the answers. I think?

Reply:
Looks good to me dude, nice find. I recon this is fine for now but it's your call if you want to change what Author does and open a PR.